presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/OptimizeMixedDistinctAggregations.java

```diff
@@ -164,20 +165,20 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<Optional<A
                                     functionCall.getWindow(),
                                     false,
                                     ImmutableList.of(aggregateInfo.getNewDistinctAggregateSymbol().toSymbolReference())),
-                            entry.getValue().getSignature(),
+                            entry.getValue().getFunctionHandle(),
                             Optional.empty()));
                 }
                 else {
                     // Aggregations on non-distinct are already done by new node, just extract the non-null value
                     Symbol argument = aggregateInfo.getNewNonDistinctAggregateSymbols().get(entry.getKey());
-                    QualifiedName functionName = QualifiedName.of("arbitrary");
-                    String signatureName = entry.getValue().getSignature().getName();
+                    QualifiedName arbitraryFunctionName = QualifiedName.of("arbitrary");
                     Aggregation aggregation = new Aggregation(
-                            new FunctionCall(functionName, functionCall.getWindow(), false, ImmutableList.of(argument.toSymbolReference())),
-                            getFunctionSignature(functionName, argument),
+                            new FunctionCall(arbitraryFunctionName, functionCall.getWindow(), false, ImmutableList.of(argument.toSymbolReference())),
+                            metadata.getFunctionManager().resolveFunction(session, arbitraryFunctionName, ImmutableList.of(new TypeSignatureProvider(symbolAllocator.getTypes().get(argument).getTypeSignature()))),
                             Optional.empty());
-                    if (signatureName.equals("count")
-                            || signatureName.equals("count_if") || signatureName.equals("approx_distinct")) {
+                    String functionName = functionCall.getName().getSuffix();
+                    if (functionName.equals("count")
```

Comment:
Even the original statement is with two line; but we can squeeze them into one in this case:
```
if (functionName.equals("count") || functionName.equals("count_if") || functionName.equals("approx_distinct")) {`
```